src/User.ts

```diff
@@ -13,50 +14,67 @@ import { Amount, RawTxObject, Signature, UserObject } from './typings'
  * related methods.
  */
 export class User {
-  private signer: TxSigner
-  private transaction: Transaction
   private provider: TLProvider
+  private signer: TLSigner
+  private transaction: Transaction
+  private wallet: TLWallet
+
+  private defaultPassword = 'ts'
 
   constructor(params: {
     provider: TLProvider
-    signer: TxSigner
+    signer: TLSigner
     transaction: Transaction
+    wallet: TLWallet
   }) {
+    this.provider = params.provider
     this.signer = params.signer
     this.transaction = params.transaction
-    this.provider = params.provider
+    this.wallet = params.wallet
   }
 
   /**
    * Checksummed Ethereum address of currently loaded user/keystore.
    */
   public get address(): string {
-    return this.signer.address
+    return this.wallet.address
```

Comment:
I think this should be signer.address, no?

Reply:
Or lets create an issue to solve that another time.

Reply:
https://github.com/trustlines-network/clientlib/issues/190